src/catalog/catalog.cpp

```diff
@@ -157,6 +157,21 @@ CatalogEntry *Catalog::GetEntry(ClientContext &context, CatalogType type, string
 	return schema->GetEntry(context, type, name, if_exists);
 }
 
+template <>
+StandardEntry *Catalog::GetEntry(ClientContext &context, string schema_name, const string &name, bool if_exists) {
```

Comment:
This method shouldn't be here in my opinion, this switching can happen one layer higher based on what type is requested (i.e. if it is an ALTER TABLE or ALTER VIEW statement).

Reply:
So I put this here b/c when I did the fetches (and exception throwing) in `bind_simple.cpp` in [this commit](https://github.com/cwida/duckdb/pull/912/commits/aff2ad8b48dfcca0cce7baced040169862e5fbb1), I got that `libc++abi.dylib: terminating with uncaught exception of type duckdb::CatalogException*
Abort trap: 6` error I mentioned in above (specifically, the unit tests that attempted to rename a non-existent table.) I didn't really understand why moving the exception from the `catalog.cpp` file to the `bind_simple.cpp` caused that to happen, but I suspect that's b/c I don't really grok C++ exceptions (the place I last wrote C++ banned them from their codebase, so I've never used them.)

Are you suggesting I create a `Catalog::GetEntry` method for the `ViewCatalogEntry` type and use that in `bind_simple.cpp` (where I'll have some sort of flag to distinguish between `ALTER TABLE` and `ALTER VIEW`)?